streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -4759,6 +4759,149 @@ public void shouldListNotPausedTasks() {
         assertEquals(taskManager.notPausedTasks().size(), 0);
     }
 
+    @Test
+    public void shouldRecyclePendingTasksFromStateDirectoryAsActive() {
+        final StandbyTask pendingTask = mock(StandbyTask.class);
+        when(pendingTask.id()).thenReturn(taskId00);
+        when(pendingTask.isActive()).thenReturn(false);
+        when(pendingTask.prepareCommit()).thenReturn(Collections.emptyMap());
+        when(pendingTask.stateManager()).thenReturn(stateManager);
+        doNothing().when(stateManager).assignToStreamThread(any(), any(), any());
+        doNothing().when(pendingTask).updateInputPartitions(any(), any());
+
+        final StreamTask activeTask = mock(StreamTask.class);
```

Comment:
You could use `StreamsTestUtils.statefulTask()` for this.

Reply:
880684e